clamav/util.py

```diff
@@ -57,22 +63,43 @@ def clamd_client():
     return client
 
 
-def scan_stream(fileobj):
+class LegacyScanResult(ScanResult):
+    def __init__(
+        self,
+        status: str,
+        signature,
+    ):
+        self.status = status
+        self.signature = signature
+
+    def malware_detected(self) -> bool:
+        return self.status != 'OK'
+
+    def finding(self):
+        return self.signature
+
+
+def scan_stream(fileobj) -> LegacyScanResult:
     c = clamd_client()
 
     result = c.instream(fileobj)
 
     if not len(result) == 1 or not 'stream' in result:
         # expected format: {"stream": (<status>, <signature-name|None>)}
         raise RuntimeError(f'result does not meet expected format: {result}')
-
     status, signature_or_none = result['stream']
-    return status, signature_or_none
+
+    return LegacyScanResult(
+        status=status,
+        signature=signature_or_none,
+    )
 
 
-def scan_container_image(image_reference: str):
+def scan_container_image(
```

Comment:
please rewrite in such a way that this function remains as it is, but split out the part of retrieving container image + layers; yield layers + path; this function is to be re-used. the actual scanning part to be "duplicated"
no need to introduce the result objects for the (soon-to-be) legacy code